src/main/scala/com/github/unchama/seichiassist/subsystems/bookedachivement/bukkit/listener/GrantBookedAchievementListener.scala

```diff
@@ -0,0 +1,47 @@
+package com.github.unchama.seichiassist.subsystems.bookedachivement.bukkit.listener
+
+import cats.data.EitherT
+import cats.effect.ConcurrentEffect
+import com.github.unchama.generic.effect.unsafe.EffectEnvironment
+import com.github.unchama.seichiassist.SeichiAssist
+import com.github.unchama.seichiassist.subsystems.bookedachivement.domain.AchievementOperation
+import com.github.unchama.seichiassist.subsystems.bookedachivement.service.AchievementBookingService
+import com.github.unchama.targetedeffect.SequentialEffect
+import org.bukkit.event.player.PlayerJoinEvent
+import org.bukkit.event.{EventHandler, Listener}
+
+class GrantBookedAchievementListener[AsyncContext[_] : ConcurrentEffect](implicit
+  effectEnvironment: EffectEnvironment,
+  service: AchievementBookingService[AsyncContext]
+) extends Listener {
+
+  import cats.effect.implicits._
+
+  @EventHandler
+  def onPlayerJoin(playerJoinEvent: PlayerJoinEvent): Unit = {
+    val player = playerJoinEvent.getPlayer
+    val uuid = player.getUniqueId
+    val playerData = SeichiAssist.playermap(uuid)
+
+    effectEnvironment.runEffectAsync("未受け取りの予約実績を読み込み付与する",
+      {
+        for {
+          ids <- EitherT(service.loadBookedAchievementsIds(player.getUniqueId))
+        } yield {
+          val effects = for {
+            (action, id) <- ids
+          } yield action match {
+            case AchievementOperation.GIVE => playerData.tryForcefullyUnlockAchievement(id)
+            case AchievementOperation.DEPRIVE => playerData.forcefullyDepriveAchievement(id)
+          }
+          effectEnvironment.runEffectAsync(
+            "実績の付与・剥奪を行う",
+            SequentialEffect(effects).run(player)
+          )
+        }
+      }
+        .value
+        .start
+    )
```

Comment:
`playerData.tryForcefullyUnlockAchievement` は[運営チームなどのメンバーがトリガーすることを想定した `TargetedEffect[CommandSender]` なので](https://github.com/GiganticMinecraft/SeichiAssist/blob/3ec3f5d4cce53a7ebc32ae5af8605e41ef946dd1/src/main/scala/com/github/unchama/seichiassist/data/player/PlayerData.scala#L804)、 `.run` する時は `.run(Bukkit.getConsoleSender)` とすると良さそうです。

また、Effectの計算のために副作用を発生させてしまっているため、

```suggestion
    val effectRunner = Bukkit.getConsoleSender
    val program = {
      for {
        ids <- EitherT(service.loadBookedAchievementsIds(player.getUniqueId))
        _ <- EitherT.right(
          ids.map {
            case (AchievementOperation.GIVE, id) =>
              playerData.tryForcefullyUnlockAchievement(id).run(effectRunner)
            case (AchievementOperation.DEPRIVE, id) =>
              playerData.forcefullyDepriveAchievement(id).run(effectRunner)
          }.sequence
        )
      } yield ()
    }.value

    effectEnvironment.runEffectAsync("未受け取りの予約実績を読み込み付与する", program.start)
```

みたいにすると尚良さそう？